### ISSUE TITLE
  Automatic refresh appears to not be working after update to 1.20.0

### ISSUE THREAD

**Issue by DotNetRob** [linuxserver/docker-freshrss#41]
[linuxserverurl]: https://linuxserver.io
[![linuxserver.io](https://raw.githubusercontent.com/linuxserver/docker-templates/master/linuxserver.io/img/linuxserver_medium.png)][linuxserverurl]

------------------------------

## Expected Behavior
Auto refresh of rss feeds should run every 30 minutes

## Current Behavior
Auto updates to appear to be working

## Steps to Reproduce

1. updated from 1.19.2 to 1.20.0 in docker
2.
3.
4.

## Environment
**OS:** Windows Docker running on windows 11
**CPU architecture:** x86_64/arm32/arm64
**How docker service was installed:** docker-compose

## Command used to create docker container (run/create/compose/screenshot)

version: "2.1"
services:
  freshrss:
    image: lscr.io/linuxserver/freshrss:latest
    container_name: freshrss
    environment:
      - PUID=1000
      - PGID=1000
      - TZ=America/Chicago
    volumes:
      - /path/to/data:/config
    ports:
      - 8585:80
    restart: unless-stopped

## Docker logs

[custom-init] No custom services found, skipping...
[custom-init] Legacy service folder /config/custom-services.d is empty, deleting...
s6-rc: info: service s6rc-oneshot-runner: starting
s6-rc: info: service s6rc-oneshot-runner successfully started
s6-rc: info: service fix-attrs: starting
s6-rc: info: service 00-legacy: starting
s6-rc: info: service 00-legacy successfully started
s6-rc: info: service fix-attrs successfully started
s6-rc: info: service legacy-cont-init: starting
cont-init: info: running /etc/cont-init.d/01-envfile
cont-init: info: /etc/cont-init.d/01-envfile exited 0
cont-init: info: running /etc/cont-init.d/01-migrations
[migrations] started
[migrations] 01-nginx-site-confs-default: executing...
/etc/cont-init.d/01-migrations: line 22: [[: found /config/nginx/site-confs/default
moving to /config/nginx/site-confs/default.conf
0: division by 0 (error token is "config/nginx/site-confs/default
moving to /config/nginx/site-confs/default.conf
0")
[migrations] 01-nginx-site-confs-default: succeeded
[migrations] 02-default-location: executing...
[migrations] 02-default-location: succeeded
[migrations] done
/etc/cont-init.d/01-migrations: line 22: [[: updating root in /config/nginx/site-confs/default.conf
0: syntax error in expression (error token is "root in /config/nginx/site-confs/default.conf
0")
cont-init: info: /etc/cont-init.d/01-migrations exited 0
cont-init: info: running /etc/cont-init.d/10-adduser

-------------------------------------
          _         ()
         | |  ___   _    __
         | | / __| | |  /  \
         | | \__ \ | | | () |
         |_| |___/ |_|  \__/

Brought to you by linuxserver.io
-------------------------------------

To support LSIO projects visit:
https://www.linuxserver.io/donate/
-------------------------------------
GID/UID
-------------------------------------

User uid:    1000
User gid:    1000
-------------------------------------

cont-init: info: /etc/cont-init.d/10-adduser exited 0
cont-init: info: running /etc/cont-init.d/11-folders
cont-init: info: /etc/cont-init.d/11-folders exited 0
cont-init: info: running /etc/cont-init.d/12-samples
cont-init: info: /etc/cont-init.d/12-samples exited 0
cont-init: info: running /etc/cont-init.d/13-nginx
Setting resolver to  127.0.0.11
Setting worker_processes to 16
cont-init: info: /etc/cont-init.d/13-nginx exited 0
cont-init: info: running /etc/cont-init.d/14-php
cont-init: info: /etc/cont-init.d/14-php exited 0
cont-init: info: running /etc/cont-init.d/15-keygen
using keys found in /config/keys
cont-init: info: /etc/cont-init.d/15-keygen exited 0
cont-init: info: running /etc/cont-init.d/20-permissions
cont-init: info: /etc/cont-init.d/20-permissions exited 0
cont-init: info: running /etc/cont-init.d/40-install
cont-init: info: /etc/cont-init.d/40-install exited 0
cont-init: info: running /etc/cont-init.d/85-version-checks
**** The following active confs have different version dates than the samples that are shipped. ****
**** This may be due to user customization or an update to the samples. ****
**** You should compare the following files to the samples in the same folder and update them. ****
**** Use the link at the top of the file to view the changelog. ****
/config/nginx/site-confs/default.conf
/config/nginx/nginx.conf

cont-init: info: /etc/cont-init.d/85-version-checks exited 0
cont-init: info: running /etc/cont-init.d/99-custom-files
[custom-init] Legacy files folder /config/custom-cont-init.d is empty, deleting...
cont-init: info: /etc/cont-init.d/99-custom-files exited 0
s6-rc: info: service legacy-cont-init successfully started
s6-rc: info: service init-mods: starting
s6-rc: info: service init-mods successfully started
s6-rc: info: service init-mods-package-install: starting
s6-rc: info: service init-mods-package-install successfully started
s6-rc: info: service init-mods-end: starting
s6-rc: info: service init-mods-end successfully started
s6-rc: info: service init-services: starting
s6-rc: info: service init-services successfully started
s6-rc: info: service legacy-services: starting
services-up: info: copying legacy longrun cron (no readiness notification)
services-up: info: copying legacy longrun nginx (no readiness notification)
services-up: info: copying legacy longrun php-fpm (no readiness notification)
s6-rc: info: service legacy-services successfully started
s6-rc: info: service 99-ci-service-check: starting
[ls.io-init] done.
s6-rc: info: service 99-ci-service-check successfully started
s6-rc: info: service 99-ci-service-check: stopping
s6-rc: info: service 99-ci-service-check successfully stopped
s6-rc: info: service legacy-services: stopping
s6-svwait: fatal: supervisor died
s6-rc: info: service legacy-services successfully stopped
s6-rc: info: service init-services: stopping
s6-rc: info: service 00-legacy: stopping
s6-rc: info: service init-services successfully stopped
s6-rc: info: service init-mods-end: stopping
s6-rc: info: service 00-legacy successfully stopped
s6-rc: info: service init-mods-end successfully stopped
s6-rc: info: service init-mods-package-install: stopping
s6-rc: info: service init-mods-package-install successfully stopped
s6-rc: info: service init-mods: stopping
s6-rc: info: service init-mods successfully stopped
s6-rc: info: service legacy-cont-init: stopping
s6-rc: info: service legacy-cont-init successfully stopped
s6-rc: info: service fix-attrs: stopping
s6-rc: info: service fix-attrs successfully stopped
s6-rc: info: service s6rc-oneshot-runner: stopping
s6-rc: info: service s6rc-oneshot-runner successfully stopped
[custom-init] No custom services found, skipping...
s6-rc: info: service s6rc-oneshot-runner: starting
s6-rc: info: service s6rc-oneshot-runner successfully started
s6-rc: info: service fix-attrs: starting
s6-rc: info: service 00-legacy: starting
s6-rc: info: service 00-legacy successfully started
s6-rc: info: service fix-attrs successfully started
s6-rc: info: service legacy-cont-init: starting
cont-init: info: running /etc/cont-init.d/01-envfile
cont-init: info: /etc/cont-init.d/01-envfile exited 0
cont-init: info: running /etc/cont-init.d/01-migrations
[migrations] started
[migrations] 01-nginx-site-confs-default: skipped
[migrations] 02-default-location: skipped
[migrations] done
cont-init: info: /etc/cont-init.d/01-migrations exited 0
cont-init: info: running /etc/cont-init.d/10-adduser
usermod: no changes

-------------------------------------
          _         ()
         | |  ___   _    __
         | | / __| | |  /  \
         | | \__ \ | | | () |
         |_| |___/ |_|  \__/

Brought to you by linuxserver.io
-------------------------------------

To support LSIO projects visit:
https://www.linuxserver.io/donate/
-------------------------------------
GID/UID
-------------------------------------

User uid:    1000
User gid:    1000
-------------------------------------

cont-init: info: /etc/cont-init.d/10-adduser exited 0
cont-init: info: running /etc/cont-init.d/11-folders
cont-init: info: /etc/cont-init.d/11-folders exited 0
cont-init: info: running /etc/cont-init.d/12-samples
cont-init: info: /etc/cont-init.d/12-samples exited 0
cont-init: info: running /etc/cont-init.d/13-nginx
cont-init: info: /etc/cont-init.d/13-nginx exited 0
cont-init: info: running /etc/cont-init.d/14-php
cont-init: info: /etc/cont-init.d/14-php exited 0
cont-init: info: running /etc/cont-init.d/15-keygen
using keys found in /config/keys
cont-init: info: /etc/cont-init.d/15-keygen exited 0
cont-init: info: running /etc/cont-init.d/20-permissions
cont-init: info: /etc/cont-init.d/20-permissions exited 0
cont-init: info: running /etc/cont-init.d/40-install
cont-init: info: /etc/cont-init.d/40-install exited 0
cont-init: info: running /etc/cont-init.d/85-version-checks
**** The following active confs have different version dates than the samples that are shipped. ****
**** This may be due to user customization or an update to the samples. ****
**** You should compare the following files to the samples in the same folder and update them. ****
**** Use the link at the top of the file to view the changelog. ****
/config/nginx/site-confs/default.conf
/config/nginx/nginx.conf

cont-init: info: /etc/cont-init.d/85-version-checks exited 0
cont-init: info: running /etc/cont-init.d/99-custom-files
[custom-init] No custom files found, skipping...
cont-init: info: /etc/cont-init.d/99-custom-files exited 0
s6-rc: info: service legacy-cont-init successfully started
s6-rc: info: service init-mods: starting
s6-rc: info: service init-mods successfully started
s6-rc: info: service init-mods-package-install: starting
s6-rc: info: service init-mods-package-install successfully started
s6-rc: info: service init-mods-end: starting
s6-rc: info: service init-mods-end successfully started
s6-rc: info: service init-services: starting
s6-rc: info: service init-services successfully started
s6-rc: info: service legacy-services: starting
services-up: info: copying legacy longrun cron (no readiness notification)
services-up: info: copying legacy longrun nginx (no readiness notification)
services-up: info: copying legacy longrun php-fpm (no readiness notification)
s6-rc: info: service legacy-services successfully started
s6-rc: info: service 99-ci-service-check: starting
[ls.io-init] done.
s6-rc: info: service 99-ci-service-check successfully started

**Comment by alexruf** [comment]
Can confirm this issue. For me auto refresh for RSS feeds also stopped working after update to container version 1.20.0

**Comment by nemchik** [comment]
From your logs:
```
**** The following active confs have different version dates than the samples that are shipped. ****
**** This may be due to user customization or an update to the samples. ****
**** You should compare the following files to the samples in the same folder and update them. ****
**** Use the link at the top of the file to view the changelog. ****
/config/nginx/site-confs/default.conf
/config/nginx/nginx.conf
```
Can you try updating these files and check it it works correctly?

**Comment by vswitch** [comment]
Same issue here as well, even after updating to the release from earlier today.

**Comment by sugarfunk** [comment]
Confirming same, no auto refresh but manually refreshing works fine.

**Comment by DotNetRob** [comment]
Apparently up until the last release the cron job was defaulted, but apparently it is not defaulted anymore.  Couldn't tell if this was on purpose or a bug.  Found multiple issues posted in the official FreshRSS Issues.   Cron job has to be manually added or included in the docker compose file.  Couldn't get it to work for the linuxserver docker-freshrss.  But got the official freshrss docker working by adding    

`environment:
      - CRON_MIN=4,34`

**Comment by nemchik** [comment]
It looks like the cron should still be running, can you try running `docker exec freshrss cat /tmp/FreshRSS.log` and posting the output here?

**Comment by sugarfunk** [comment]
> It looks like the cron should still be running, can you try running `docker exec freshrss cat /tmp/FreshRSS.log` and posting the output here?

/bin/ash: /usr/bin/php7: not found

**Comment by shimeike** [comment]
The `crontab` present in the image looks to be incorrect.

The relevant auto-update entry attempts to run php7 (which no longer appears to be present in this alpine image). Additionally, the `actualize_script.php` is not present in the location where the crontab is expecting to find it.

I think I fixed my container by editing the relevant `crontab` command to:
`/usr/bin/php /app/www/app/actualize_script.php > /tmp/FreshRSS.log 2>&1`

**Comment by mztiq** [comment]
I can confirm the fix from @shimeike is working, the path to the `actualize_script.php` has changed and therefore crontab needs to be updated.

**Comment by nemchik** [comment]
I see the issue now!

The upgrade to the image did not automatically replace the line in existing user cron files. I can prepare a fix today.

If you haven't customized your cron file you can delete it and restart the container and a new working file will be created.

Later today I will add in a fix to replace existing user cron to have the correct line.

**Comment by sugarfunk** [comment]
Renamed the file "root" to "root.old" in Cron tab rebooted and all good now.

**Comment by Cnote43** [comment]
I'm still having this problem. Oddly, one of my 35 feeds refreshes automatically, but not the other 34. I've tried:
- Renaming the root crontab file to root.old and restarting the container. It recreated the root file with the original php7 line.
- Manually changing the cron line to `/usr/bin/php /app/www/app/actualize_script.php > /tmp/FreshRSS.log 2>&1`. This did nothing.

**Comment by nemchik** [comment]
> docker exec freshrss cat /tmp/FreshRSS.log

Try running
```shell
docker exec freshrss ls /app/www/app/actualize_script.php
```

If that gives an error then you have not updated to the latest image, and your issue may be something else entirely.

Either way, you could run `docker exec freshrss cat /tmp/FreshRSS.log` and post the output here and I may be able to help.

**Comment by Cnote43** [comment]
@nemchik , thanks for the offer to help. The actualize script is in /app/www/app. I ran `docker exec freshrss cat /tmp/FreshRSS.log` and it showed a successful run of the actualize script with the timestamp reflecting when I ran that command. I've given it a few hours now and it seems to be refreshing all feeds automatically. I'll sit on it over the weekend and report back if it stops working. Thank you!
